src/query/block/meta.go

```diff
@@ -127,6 +210,18 @@ func combineWarnings(a, b Warnings) Warnings {
 	return nil
 }
 
+func combineMetricMetadata(a, b map[string]*ResultMetricMetadata) map[string]*ResultMetricMetadata {
+	if a == nil && b == nil {
+		return nil
+	}
+
+	merged := make(map[string]*ResultMetricMetadata)
```

Comment:
Is there a reason we're creating a new one instead of using the `mergeMetricMetadataMaps` to merge a into b?

Reply:
There's an existing pattern for `Combine` though I agree that `merge` probably makes more sense.

Reply:
nit: init this thing with size (can do like max of len(a) or len(b) for a decent starting point?)